src/profile-logic/symbolication.js

```diff
@@ -224,15 +246,16 @@ export function applyFunctionMerging(
 }
 
 /**
- * Symbolicate the given thread. Calls cbo.onMergeFunctions and cb.onGotFuncNames after each bit of
- * symbolication, and resolves the returned promise once completely done.
- * @param  {Object}  thread      The thread to symbolicate, in the "processed profile" format.
- * @param  {Number}  threadIndex The index of the thread. This is needed because it gets supplied in the callbacks.
- * @param  {Object}  symbolStore A SymbolStore object that can be used for getting the required symbol tables.
- * @param  {Object}  cbo         An object containing callback functions 'onMergeFunctions'  and 'onGotFuncNames'.
- * @return {Promise}             A promise that resolves (with nothing) once symbolication of the thread has completed.
+ * Symbolicate the given thread. Calls the symbolication handlers `onMergeFunctions`
+ * and `onGotFuncNames` after each bit of symbolication, and resolves the returned
+ * promise once completely done.
  */
-function symbolicateThread(thread, threadIndex, symbolStore, cbo) {
+function symbolicateThread(
+  thread: Thread,
+  threadIndex: ThreadIndex,
+  symbolStore: SymbolStore,
+  symbolicationHandlers: SymbolicationHandlers
+): Promise<void[]> {
```

Comment:
I think this could also be done simply by making this function an async function and changing the top level `return` into `await` :)

The map function could likely be made async too but I think this is too much of a change for this patch.